pkg/skaffold/runner/runner.go

```diff
@@ -76,9 +79,21 @@ type SkaffoldRunner struct {
 	defaultLabeller      *deploy.DefaultLabeller
 	portForwardResources []*latest.PortForwardResource
 	builds               []build.Artifact
-	hasBuilt             bool
-	hasDeployed          bool
 	imageList            *kubernetes.ImageList
+
+	hasBuilt    bool
+	hasDeployed bool
+
+	buildIntent  bool
+	syncIntent   bool
+	deployIntent bool
+
+	// NOTE: these reset functions require the intentLock to be acquired before calling.
+	resetBuildIntent  func()
+	resetSyncIntent   func()
+	resetDeployIntent func()
+
+	intentLock *gosync.Mutex
```

Comment:
I would feel much better if those 7 fields were extracted to an `intents` type. The SkaffoldRunner is already pretty crowded and it might help simplify the rest of the code.

Reply:
good point, done